File: WebServices/app/src/main/java/com/catherine/webservices/network/DownloaderListener.java
package com.catherine.webservices.network;

/**
 * Created by Catherine on 2017/11/29.
 * Soft-World Inc.
 * catherine919@soft-world.com.tw
 */

public interface DownloaderListener {
    void update(int threadID, int downloadedLength, int LENGTH);

    void connectFailure(HttpResponse response, Exception e);
}

File: app/src/main/java/com/catherine/webservices/network/DownloaderListener.kt
package com.catherine.webservices.network

/**
 * Created by Catherine on 2017/8/28.
 * Soft-World Inc.
 * catherine919@soft-world.com.tw
 */
interface DownloaderListener {
    fun update(downloadedLength: Int, LENGTH: Int)
    fun connectFailure(response: HttpResponse, e: Exception?)
}